README.md

```diff
@@ -32,6 +32,28 @@ To provide values for these configuration values, they must be either:
 **OR**
 * Passed as compile time preprocessor macros
 
+## Porting the coreMQTT Agent Library
+In order to use the MQTT Agent library on a platform, you need to supply thread safe functions for the agent's [messaging interface](source/include/core_mqtt_agent_message_interface.h).
+
+### Messaging Interface
+Each of the following functions must be thread safe.
+
+| Function Pointer | Description |
+| :-: | --- |
+| `MQTTAgentMessageSend_t` | A function that sends `MQTTAgentCommand_t *` pointers to be received by `MQTTAgent_CommandLoop`. This can be implemented by pushing to a thread safe queue. |
+| `MQTTAgentMessageRecv_t` | A function used by `MQTTAgent_CommandLoop` to receive `MQTTAgentCommand_t *` that were sent by API functions. This can be implemented by receiving from a thread safe queue. |
+| `MQTTAgentCommandGet_t` | A function that returns a pointer to an allocated `MQTTAgentCommand_t` structure, to be used for information and arguments for a command to be executed by `MQTTAgent_CommandLoop()`. If using dynamic memory, this can be implemented using `malloc()`. |
```

Comment:
```suggestion
| `MQTTAgentCommandGet_t` | A function that returns a pointer to an allocated command object (of `MQTTAgentCommand_t` type) that is executed by the MQTT Agent command loop. If using dynamic memory, this can be implemented using `malloc()`. |
```